### ISSUE TITLE
popup.pollPopup promise does not fire on facebook

### ISSUE THREAD

**Issue by nikopolidi** [sahat/satellizer#721]
Hello guys I'm experiencing an issue which is hanging me.
I'm trying to to implement facebook authentication but I'm not getting result from frame.
env versions:
angular#1.4.4
satellizer#0.13.4 version

app and server running on different servers so for example here is local development hosts
```
API: http://localhost:1338
APP: http://localhost:9000
```
I've implemented facebook authentication on server side with [machinepack-facebook](https://www.npmjs.com/package/machinepack-facebook) and it supposed to work good.
I have 2 endpoints for that
```
http://localhost:1338/api/v2/auth/facebooklogin
http://localhost:1338/api/v2/auth/facebookcallback 
```
where expected response from /facebookcallback is
```
// server code
res.send({token: ""JWT_TOKEN:HERE})
// response body
{
token: "JWT_TOKEN_HERE"
}
```
and have configured $authProvider on client-side
```
$authProvider.tokenName = 'token';
$authProvider.baseUrl = apiEndPoint
$authProvider.facebook({
    clientId: MY_CLIENT_ID
    ,url: apiEndPoint+'/auth/facebooklogin'
    ,redirectUri: apiEndPoint+"/auth/facebookcallback"
    ,type: '2.0'
  });
```

So basically when I'm calling 
```
$auth.authenticate('facebook')
```
I'm not able to access any of that promise result
after logging in to facebook there is correct response from server displayed at frame window
```
{
token: "JWT_TOKEN_HERE"
}
```
and no promise results.

Please guys help me with that issue. I will provide you any details you need to understand whats wrong.
Thanks!

**Comment by sahat** [comment]
Hi @nikopolidi,

`redirectUri` is the URL where Facebook will redirect to, with the `code` parameter. This is all happening within popup window. Actually where you redirect to, is almost irrelevant, because as soon as it sees `code` parameter, the popup will be closed.

After popup is closed it will send that `code` param to your `url` that you specified in the config.
```js
return openPopup
.then(function(oauthData) {
  // When no server URL provided, return popup params as-is.
  // This is for a scenario when someone wishes to opt out from
  // Satellizer's magic by doing authorization code exchange and
  // saving a token manually.
  if (defaults.responseType === 'token' || !defaults.url) {
    return oauthData;
  }

  if (oauthData.state && oauthData.state !== storage.get(stateName)) {
    return $q.reject('OAuth "state" mismatch');
  }
  // this line sends `code` to your server
  return Oauth2.exchangeForToken(oauthData, userData);
});
``` 

I don't know specifics of how machinepack auth works. What happens when you redirect to `/auth/facebookcallback` with `code` param? Does it do the entire auth flow similar to what I have in my repo? If that's the case, I am guessing it is sort of fighting with `Oauth2.exchangeForToken`. Because that function continues the promise flow. If it fails, the whole flow fails, even if you have already obtained access token / JWT from your /auth/callback route.

Notice that `if-statement` above that checks if you have not provided `url`. Try leaving out `url` property and see if it helps.
